File: two-pointers/0088-merge-sorted-array/merge-sorted-array-solution.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        nums1.erase(nums1.begin()+m,nums1.end());
        for(auto e:  nums2)
            nums1.push_back(e);
        sort(nums1.begin(),nums1.end());
    }
};

File: two-pointers/0088-merge-sorted-array/merge-sorted-array-solution.java
import java.util.Arrays;
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int index2 = n - 1;
        int index1 = m - 1;
        int bigIndex = m + n - 1;

        while (index2 >= 0 && index1 >= 0) {
            if (nums1[index1] <= nums2[index2]) {
                nums1[bigIndex] = nums2[index2];
                index2--;
            } else {
                nums1[bigIndex] = nums1[index1];
                index1--;
            }
            bigIndex--;
            
        }
        while (index2 >= 0) {
            nums1[bigIndex--] = nums2[index2--];
        }
        
    }
}
